src/components/Jam.jsx

```diff
@@ -46,34 +48,28 @@ export default function Jam() {
   const [collaborativeOperationRunning, setCollaborativeOperationRunning] = useState(false)
   const [schedule, setSchedule] = useState(null)
 
+  // Returns one fresh address for each requested mixdepth.
   const getNewAddresses = useCallback(
-    (count, mixdepth) => {
+    (mixdepths) => {
+      if (mixdepths.length !== 3) {
+        throw new Error('Can only handle 3 destination addresses for now.')
+      }
       if (!walletInfo) {
         throw new Error('Wallet info is not available.')
       }
-      const externalBranch = walletInfo.data.display.walletinfo.accounts[mixdepth].branches.find((branch) => {
-        return branch.branch.split('\t')[0] === 'external addresses'
-      })
-
-      const newAddresses = []
+      return mixdepths.map((mixdepth) => {
+        const externalBranch = walletInfo.data.display.walletinfo.accounts[mixdepth].branches.find((branch) => {
+          return branch.branch.split('\t')[0] === 'external addresses'
+        })
 
-      externalBranch.entries.every((entry) => {
-        if (entry.status === 'new') {
-          newAddresses.push(entry.address)
-        }
+        const newEntry = externalBranch.entries.find((entry) => entry.status === 'new')
 
-        if (newAddresses.length >= count) {
-          return false
+        if (!newEntry) {
```

Comment:
Just a small notice here. If the exception is thrown, the user interface behaves very weird. Maybe try it for yourself. This is not something that should ever happen, but maybe try to wrap all invocations in a try/catch block if it can be mitigated. What do you think?